Build error: "error: ‘uintptr_t’ was not declared in this scope"

## Error
```
[  0%] Built target Luau.Common
[  0%] Building CXX object CMakeFiles/Luau.Ast.dir/Ast/src/Ast.cpp.o
In file included from /home/radiatedexodus/LocalSources/luau/Ast/src/Ast.cpp:2:
/home/radiatedexodus/LocalSources/luau/Ast/include/Luau/Ast.h: In member function ‘std::size_t std::hash<Luau::AstName>::operator()(const Luau::AstName&) const’:
/home/radiatedexodus/LocalSources/luau/Ast/include/Luau/Ast.h:1312:17: error: ‘uintptr_t’ was not declared in this scope
 1312 |         return (uintptr_t(value.value) >> 4) ^ (uintptr_t(value.value) >> 9);
      |                 ^~~~~~~~~
/home/radiatedexodus/LocalSources/luau/Ast/include/Luau/Ast.h:11:1: note: ‘uintptr_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
   10 | #include <string.h>
  +++ |+#include <cstdint>
   11 |
make[3]: *** [CMakeFiles/Luau.Ast.dir/build.make:76: CMakeFiles/Luau.Ast.dir/Ast/src/Ast.cpp.o] Error 1
make[2]: *** [CMakeFiles/Makefile2:135: CMakeFiles/Luau.Ast.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:306: CMakeFiles/Luau.Repl.CLI.dir/rule] Error 2
make: *** [Makefile:215: Luau.Repl.CLI] Error 2
```

## Reproduction steps
1. ``git clone https://github.com/Roblox/luau.git``
2. ``cd luau``
3. ``mkdir build``
4. ``cd build``
5. ``cmake .. -DCMAKE_BUILD_TYPE=Release``
6. ``cmake --build . --target Luau.Repl.CLI Luau.Analyze.CLI Luau.Ast.CLI Luau.Reduce.CLI --config Release``
7. Error occurs

## CMake generated config logs
```
-- The CXX compiler identification is GNU 13.1.1
-- The C compiler identification is GNU 13.1.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/sbin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/sbin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Configuring done (0.3s)
-- Generating done (0.0s)
-- Build files have been written to: /home/radiatedexodus/LocalSources/luau/build
```

## System
x86_64 Arch Linux (inside Windows Subsystem for Linux 2)
I have confirmed I have ``cmake``, ``gcc``, and ``clang`` installed

--- a/Ast/include/Luau/Ast.h
+++ b/Ast/include/Luau/Ast.h
@@ -8,6 +8,7 @@
 #include <string>
 
 #include <string.h>
+#include <stdint.h>
 
 namespace Luau
 {

--- a/Ast/src/StringUtils.cpp
+++ b/Ast/src/StringUtils.cpp
@@ -7,6 +7,7 @@
 #include <vector>
 #include <string>
 #include <string.h>
+#include <stdint.h>
 
 namespace Luau
 {

